### DISCUSSION TITLE
German Language

### DISCUSSION THREAD

**Discussion by ingo1278** [UnchartedBull/OctoDash#2271] [General]
is it possible to set the language to german?

thanks

**Comment by jneilliii** [comment]
I think it is. IIRC you have to configure your pi's localization options and a translation file has to be available.

**Reply by jneilliii** [reply]
https://github.com/UnchartedBull/OctoDash/issues/914#issuecomment-891214835

**Comment by ingo1278** [comment]
How can i change the Language? I changed the localization to german. But the Language are english

**Reply by jneilliii** [reply]
You have to install that one linked earlier that has the language integrated.

**Reply by ingo1278** [reply]
How Can i copy the deb file to the pi?

**Reply by jneilliii** [reply]
I personally use filezilla ftp client, using ssh connection options, but you can also use winscp.

**Reply by ingo1278** [reply]
i saw that the version is an older one.

**Comment by ingo1278** [comment]
How can I install the latest version with German support?

**Reply by jneilliii** [reply]
That one linked in that thread should be the latest version with German support. Did it report a different version? The command to install is `sudo dpkg -i ./octodash_2.2.0_armv7l-de.deb` from the folder where you transferred it to.

**Reply by ingo1278** [reply]
There is a newer version that is only in English. I don't want to install an old version, just because of the German language. It should be possible to always have the latest version with German text

**Reply by jneilliii** [reply]
not an officially released newer version, just ones that have been compiled for people to test spool manager integration best I know.

**Reply by ingo1278** [reply]
Nowadays it should be possible to translate the texts yourself via a text file. that's really bad with octodash

**Reply by UnchartedBull** [reply]
The main issue is that the german translation file isn't up to spec. Please feel free to edit them yourself: https://github.com/UnchartedBull/OctoDash/blob/bd34f1a25997d872c2af6e97121ce8bd91ec1aab/src/locale/messages.de.xlf. Once this has been merged into main it will be included in all future releases (and all unofficial builds from that commit on). My main gripe here is that I would prefer "du" instead of "sie" and some sentences are a bit formal. I haven't had the time yet to go over the file unfortunately.

Also translating a build application via a text file will never be possible with OctoDash, since there are many optimisations going on when everything is build to reduce load on the Pi. So only language files that have been present at build time will be included.

**Comment by ingo1278** [comment]
Timon G. wrote to me that I should edit the messages.de.xlf file. I can do it, but I can't edit / save it. How can i do this? And why is the existing language file not included in the current versions?

**Reply by UnchartedBull** [reply]
You would need to fork the repository and then you can edit that file on your local repository. Alternatively if you want to skip the whole git thing you can just download the file and edit it with your favourite local text editor and then just put it somewhere here. I'll add that to the repo then.

The file isn't included since neither me nor the original author were satisfied with that, so we chose to wait until it's good to be merged.

**Reply by ingo1278** [reply]
ok I try it to translate

**Reply by ingo1278** [reply]
I translated it and inserted the file in the post (See next Post)

**Comment by ingo1278** [comment]
here I translate the text in German. I hope, that file you are insert into the setup. Can you tell me, when you done? please remove xlsx at the end
[messages.de.xlf.xlsx](https://github.com/UnchartedBull/OctoDash/files/7213136/messages.de.xlf.xlsx)

**Reply by jneilliii** [reply]
If you want to verify I added in your translation and updated the other necessary parts in my manual_build branch. Go to the link below and in theory the artifact from that page can be extracted and you can transfer the arm deb file and run the install command. 

https://github.com/jneilliii/OctoDash/actions/runs/1264267275

once confirmed I could submit a PR to OctoDash, assuming @UnchartedBull is good with the translation.

**Reply by ingo1278** [reply]
What should I do now? Wait for @UnchartedBull ?

**Comment by ingo1278** [comment]
How long does it go until the German language is in the update

**Reply by UnchartedBull** [reply]
Will be part of the next release. There is no due date set yet for the next release, but shouldn't be too long hopefully.
